### PR TITLE
Remove redundant running containers of pod

### LINKED ISSUE #66186: redundant running container exists
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
docker restart container of Pod
some times,there will be 2 running containers exits

**What you expected to happen**:
it should be only one running containers

**How to reproduce it (as minimally and precisely as possible)**:
1. Start a pod with one running container (exclude the pause container)
2. Run cmd: docker stop xxxxxx_0 , to stop the container, and kubelet will create a new container name xxxxxx_1
3. Run cmd: docker start xxxxx_0, now there are 2 running containers, one is xxxxxx_0, and one is xxxxxx_1

If we repeat step 2 and 3, there may be more than 2 running container

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):1.10.5
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by linyouchong**
**What this PR does / why we need it**:
Remove redundant running containers of pod

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #66186

**Special notes for your reviewer**:
NONE

**Release note**:
```release-note
kubelet: Remove redundant running containers of pod
```

### CODE REVIEW

File: pkg/kubelet/kuberuntime/kuberuntime_manager.go [hunk 1]
```diff
@@ -553,6 +558,29 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 		changes.KillPod = true
 	}
 
+	// compute containers to be killed
+	runningContainerStatuses := podStatus.GetRunningContainerStatuses()
+	for _, containerStatus := range runningContainerStatuses {
+		_, keep := containersToKeep[containerStatus.ID]
+		if !keep {
```
Comment (neolit123) [inline pkg/kubelet/kuberuntime/kuberuntime_manager.go]: you can do this to avoid indenting the whole block:
```
if keep {
	continue
}
```

**Comment by neolit123** [comment]
i would change the release note to:
> kubelet: Remove redundant running containers of pod

**Comment by linyouchong** [comment]
@neolit123 fixed, PTAL
Thanks for your review

**Comment by neolit123** [comment]
@linyouchong 

thanks for the release note update.
`/pkg/kubelet/kuberuntime/kuberuntime_manager.go` this file needs a gofmt update to make the CI happy.

**Pull request closed without merging**